OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -40,12 +41,16 @@ public class ArmamentInfo : PausableConditionalTraitInfo, Requires<AttackBaseInf
 		public readonly int FireDelay = 0;
 
 		[Desc("Muzzle position relative to turret or body, (forward, right, up) triples.",
-			"If weapon Burst = 1, it cycles through all listed offsets, otherwise the offset corresponding to current burst is used.")]
+			"If weapon Burst = 1, it cycles through all listed offsets, otherwise the offset corresponding to current burst is used.",
+			"Every position here will generate a barrel in this armament.")]
 		public readonly WVec[] LocalOffset = Array.Empty<WVec>();
 
-		[Desc("Muzzle yaw relative to turret or body.")]
+		[Desc("Muzzle yaw relative to turret or body on generated barrel.")]
 		public readonly WAngle[] LocalYaw = Array.Empty<WAngle>();
 
+		[Desc("Allow player to set the barrel index manually if needed.")]
+		public readonly int[] BarrelsIndex = Array.Empty<int>();
```

Comment:
Why is this needed exactly? I don't see a point in in setting it manually

Reply:
This is will provide a convenient YAML writing for something like:

1. Fire at the same time
```
Armament@1:
    Weapon: cannon1
    Name: primary
    BarrelsIndex: 0, 1
Armament@2:
    Weapon: cannon1
    Name: primary
    BarrelsIndex: 1, 0
ThrowsCasing: 
    Armaments: primary
    Weapon: casing1
    CasingOffset: [offset0], [offset1]
```
Or you have to write 2 `ThrowsCasing` trait.

2. More than one fire effect for one Armament
```
Armament@1:
    Weapon: MRLSlauncher1
    Name: primary
    BarrelsIndex: 1, 0, 1, 0
ThrowsCasing@1: 
    Armaments: primary
    Weapon: casing1
    CasingOffset: [offset0], [offset1]
ThrowsCasing@2: 
    Armaments: primary
    Weapon: casing2
    CasingOffset: [offset0], [offset1]
```
Or you have to write  [offset0], [offset1],  [offset3], [offset4] for each `ThrowsCasing`

Reply:
Oh I see, this is for firing multiple barrels in the same frame. But wouldn't Burst: 2 BurstDelay 0 achieve the same result and be much clearer to modders?

Reply:
Regretably, no. "BurstDelay=0" will still result in burst fire at the next frame. I filled up #20362 to achieve this but I have no idea on making it work normal with `AmmoPool`

Reply:
The should make it work with 0, or think of another solution. The `BarrelsIndex:` implementation feels more like a hack and is fairly confusing. You have to really study the code to understand how it works

Reply:
Yes and this one is tricky, but it brings some prefs and code simplify.